vortex-array/src/compute/slice.rs

```diff
@@ -45,7 +46,12 @@ pub fn slice(array: impl AsRef<Array>, start: usize, stop: usize) -> VortexResul
 
     check_slice_bounds(array, start, stop)?;
 
-    let derived_stats = derive_sliced_stats(array);
+    // We know that constant array don't need stats propagation, so we can avoid the overhead of
+    let derived_stats = if ConstantArray::maybe_from(array).is_some() {
```

Comment:
```suggestion
    let derived_stats = if array.is_constant() {
```

Reply:
I only want to apply this if I can check that the array is constant with zero cost